src/answers-umd.js

```diff
@@ -168,6 +169,12 @@ class Answers {
       console.error(`Context parameter "${context}" is invalid, omitting from the search.`);
     }
 
+    if (globalStorage.getState(StorageKeys.REFERRER_PAGE_URL) === null) {
+      const referrer = urlWithoutQueryParamsAndHash(document.referrer || '');
+      persistentStorage.set(StorageKeys.REFERRER_PAGE_URL, referrer, true);
```

Comment:
Should we even set a value in persistent/global storage if there is no document.referrer?

Reply:
I suppose the `||''` on L173 is unnecessary, since `document.referrer` should always be on the window. I added that mostly in case we add ANSWERS.init tests that don't not have a `window.document`. What do you think?

Reply:
I think we can cross that bridge when we come to it. For now, I'd rather we not set anything if there is no referrer. Conceptually, that feels a bit better to me.

Reply:
Could you explain the reasoning here so I know for next time? If there is no referrer, we want the ANSWERS.init to fail? Or we just don't want to assume the document.referrer if it doesn't exist?

Reply:
If there's no referrer, we don't want Answers.init to fail. I don't think there will always be a referrer. I'd say we should not set anything in globalStorage or persistentStorage at all in this case, since there's not really anything to store.

Reply:
On second thought, might be worth asking the PMs about this situation. If there is no referrer, should we default to no query param or use the current URL? If someone were to bookmark a site and use that to navigate to it, for instance, there'd be no referrer.

Reply:
Hello! If there is no referrer, we want that to be explicit in the URL. The PM stressed that we want the referrer to be carried within an iFrame such that if you navigate to a different tab or generally within the experience, then the referrer is still the same. Thus, if it is empty, we should explicitly say it is empty so that is carried (otherwise we would get the home page as the referrer when we navigate with the tabs, for example).

Did I understand that correctly?

Reply:
If this is a question about what value we should use to determine if someone came to the page directly, I'd be curious about why we would want to stray from the `document.referrer` value.

I asked by giving an example a while ago with `''` as the referrer and the PM was good with it, just confirmed again and they said it works! How do you feel?

Reply:
Okay, so if they navigated to the Answers page directly, we'd want to have a 'referrerPageUrl=' in the URL? Then that would be persisted across all pages of the experience (or iframe). Based on this: https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer, I think the typeof will always be string. So, that check might be redundant. Once we get rid of that, seems good to me!

Reply:
Ah, misunderstanding here. When I asked "when there is no referrer", I should have specified I meant "null/undefined" not `""`. Changed!